docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,289 @@
 
 ```
+// tcp_client_listener.cc
+// Code has been simplified and whitespace has been added for readability.
 
-#include "transport_manager.h"
-#include "transport_manager_default.h"
+TransportAdapter::Error TcpClientListener::StopListening() {
+  if (!started_) {
+    LOG("TcpClientListener is not running now");
+    return TransportAdapter::BAD_STATE;
+  }
 
-class MyTransportAdapterListener : public TransportAdapterListenerImpl
-{
-    // customize listener
-}
+  interface_listener_->Stop();
 
-...
-
-{
-    // The default implementation of TCP adapter and the default implementation of adapter listener are used in this example
-    // Developer is free to create his own implementation from scratch
-    TransportAdapter *my_transport_adapter = static_cast<TransportAdapter*>(new TcpTransportAdapter);
-    my_transport_adapter->init();
-    TransportManager *tm_impl = TransportManagerDefault::instance();
-    TransportAdapterListener *my_ta_listener = new MyTransportAdapterListener(tm_impl);
-    my_transport_adapter->addListener(my_ta_listener);
-    tm_impl->addTransportAdapter(my_transport_adapter);
-}
+  StopListeningThread();
 
+  started_ = false;
+  LOG("Tcp client listener was stopped successfully");
+  return TransportAdapter::OK;
+}
 ```
 
-## Creating a TM with custom TAs only
+The StopListening() and SuspendListening() functions do about the opposite, both stop the TCP Client Listener delegate thread and then set `started_` to false. The difference between the two functions on the TCP Client Listener is that StopListening() will also stop the Platform Specific Network Interface Listener's delegate thread. When SuspendListening() is called, SDL will not be able to create new connections, but existing connections are still able to communicate data. StopListening() will also kill communication with existing connections.
 
-If for some reason the default adapters are not a good fit they can be completely replaced with developer defined adapters. To do this the developer
+#### Server Connection Factory
 
-  1. Must implement the TA logic
-  2. Provide the new TA to the TM
+The Server Connection Factory has the method CreateConnection() which, provided with a device UID and application handle, creates a connection to the application, and then should call ConnectionCreated() on the Transport Adapter.
 
 ```
+// tcp_connection_factory.cc
+// Code has been simplified and whitespace has been added for readability.
 
-#include "transport_manager.h"
-#include "transport_manager_impl.h"
+TransportAdapter::Error TcpConnectionFactory::CreateConnection(
```

Comment:
I would pick another transport type to explain the CreateConnection. I could be wrong but I dont think the TCPConnectionFactory is ever used?